pkg/daemon/ceph/agent/controller.go

```diff
@@ -0,0 +1,133 @@
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package agent
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/daemon/ceph/agent/flexvolume"
+	opagent "github.com/rook/rook/pkg/operator/ceph/agent"
+	opcontroller "github.com/rook/rook/pkg/operator/ceph/controller"
+	"github.com/rook/rook/pkg/operator/ceph/provisioner"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	flexcontroller "sigs.k8s.io/sig-storage-lib-external-provisioner/v6/controller"
+)
+
+const (
+	controllerName        = "rook-ceph-operator-flex-controller"
+	provisionerName       = "ceph.rook.io/block"
+	provisionerNameLegacy = "rook.io/block"
+)
+
+var (
+	// The supported configurations for the volume provisioner
+	provisionerConfigs = map[string]string{
+		provisionerName:       flexvolume.FlexvolumeVendor,
+		provisionerNameLegacy: flexvolume.FlexvolumeVendorLegacy,
+	}
+)
+
+// ReconcileAgent reconciles the Flex driver
+type ReconcileAgent struct {
+	client   client.Client
+	context  *clusterd.Context
+	opConfig opcontroller.OperatorConfig
+}
+
+// Add creates a new CephClient Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, context *clusterd.Context, opManagerContext context.Context, opConfig opcontroller.OperatorConfig) error {
+	return add(mgr, newReconciler(mgr, context, opConfig))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, context *clusterd.Context, opConfig opcontroller.OperatorConfig) reconcile.Reconciler {
+	return &ReconcileAgent{
+		client:   mgr.GetClient(),
+		context:  context,
+		opConfig: opConfig,
+	}
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	logger.Infof("%s successfully started", controllerName)
+
+	// Watch for ConfigMap (operator config)
+	err = c.Watch(&source.Kind{
+		Type: &v1.ConfigMap{TypeMeta: metav1.TypeMeta{Kind: "ConfigMap", APIVersion: v1.SchemeGroupVersion.String()}}}, &handler.EnqueueRequestForObject{}, predicateController())
```

Comment:
can we split this to multiple lines for me it's easier to read when it is multi-line?
same for other places.

Reply:
Sure can do!